client/allocrunner/taskrunner/service_hook.go

```diff
@@ -199,6 +199,10 @@ func (h *serviceHook) getTaskServices() *agentconsul.TaskServices {
 // interpolateServices returns an interpolated copy of services and checks with
 // values from the task's environment.
 func interpolateServices(taskEnv *taskenv.TaskEnv, services []*structs.Service) []*structs.Service {
+	if taskEnv == nil || len(services) == 0 {
```

Comment:
The task env is not set in the constructor (https://github.com/hashicorp/nomad/blob/a5ba15591a90a83630205054de8533cc0950a48e/client/allocrunner/taskrunner/service_hook.go#L52) so if Killing or Exited is called before post start/update it would panic

Reply:
Add a comment